matmeta/payload_metaclass.py

```diff
@@ -336,8 +336,27 @@ def _add_source(self, metadata):
 
     def _add_people(self, metadata):
         people = []
-        
+        # use full names instead of ids to determine whether to add a new entry
+        # TODO: find a more elegant solution
+        def person_already_added(new_citrine_person):
+            for citrine_person in people:
+                if (
+                    citrine_person.name.family == new_citrine_person.name.family
+                    and 
+                    citrine_person.name.given == new_citrine_person.name.given
+                ):
+                    if new_citrine_person.tags:
+                        new_tags = list(set(new_citrine_person.tags))
+                        if citrine_person.tags:
+                            old_tags = list(set(citrine_person.tags))
+                            citrine_person.tags = old_tags.extend(new_tags)
+                        else:
+                            citrine_person.tags = new_tags
+                    return True 
+            return False
+
         def add_to_people(person_list, tags):
+            if tags: tags = list(tags)

```

Comment:
What if `tags` is already a list?

Reply:
Tags can be any iterable.  If it is already a list, this code will simply make a defensive copy.  It could also be written tags = tags[:].